psutil/_psutil_osx.c

```diff
@@ -1798,6 +1799,57 @@ psutil_cpu_stats(PyObject *self, PyObject *args) {
 }
 
 
+/*
+ * Return temperatures of hardware components.
+ */
+static PyObject *
+psutil_sensors_temperatures(PyObject *self, PyObject *args) {
+    PyObject *py_retdict = PyDict_New();
+    // CPU only stats
+    PyObject *py_cpu_temp = PyDict_New();
+    PyObject *py_cpu_temp_current = NULL;
+    PyObject *py_cpu_temp_high = NULL;
+
+    if (py_retdict == NULL)
+        goto error;
+
+    if (py_cpu_temp == NULL)
+        goto error;
+
+    py_cpu_temp_current = Py_BuildValue("d", SMCGetTemperature(SMC_KEY_CPU_TEMP));
+    if (!py_cpu_temp_current)
+        goto error;
+
+    if (PyDict_SetItemString(py_cpu_temp, "current", py_cpu_temp_current)) {
+        goto error;
+    }
+
+    py_cpu_temp_high = Py_BuildValue("d", SMCGetTemperature(SMC_KEY_CPU_TEMP_HIGH));
+    if (!py_cpu_temp_high)
+        goto error;
+
+    if (PyDict_SetItemString(py_cpu_temp, "high", py_cpu_temp_high)) {
+        goto error;
+    }
+    if (PyDict_SetItemString(py_retdict, "TC0F", py_cpu_temp)) {
+        goto error;
+    }
+
+
+    Py_DECREF(py_cpu_temp);
+    Py_DECREF(py_cpu_temp_current);
+    Py_DECREF(py_cpu_temp_high);
+    return py_retdict;
```

Comment:
I think it's easier if you use a list instead of a dict, and append values as you find them;